docs/docs/spec.md

```diff
@@ -1038,6 +1038,21 @@ The Java rules are defined in [JLS 5.1.10]. We add to them as follows:
 
     > See ["Augmented null types."](#null-types)
 
+## Harmonizing annotations on special type usages {#harmoize-annotations}
```

Comment:
I'm expecting The Beach Boys to jump in here. (_Ba-ba-ba-ba-ba-Barbara Annotation!_)

If this bit is really speciifc to `equals(Object)`'s parameter in record classes, then lets name this section more specifically: no harmonization, just "Record class `equals` parameter" or something.

On the other hand, we could make at least this bit a little more general, saying that in any case where an explicit type usage in code is nullable regardless of annotation, then it can't have `@NonNull` and it must have `@Nullable` in null-marked code or in all-worlds checkers. Then, non-normatively, we could point out that the only such case is the special one for record class `equals` parameters.

Reply:
I had hoped to find a title to distinguish this section from the _other_ discussion of record `equals` methods above. But then I had gotten worried that I'd end up using terminology that suggests that we're mandating user-facing tool behavior :) I ended up here, with a title that attempts to suggest that it's a piece of a larger puzzle.

But in the end, I also had to settle on some phrasing for the bullet points under this section, and I decided I could live with "expected." That's a little more generic than I'd aim for in a section title, since people could consider various annotations to be "unexpected" or "unexpected" in various cases. But I think it's fine, especially with an explicit mention of record `equals` methods (and a link back to the _other_ section about record `equals` methods to emphasize that it's not the whole story).

So:

#### Expected annotations on record `equals` methods